Doesn't build on M1 Macbook macOS

Might be worth mentioning in readme or adding support. The error I get is:
```
> nix run diamondburned/nix-search/v0.3.1
error: builder for '/nix/store/0gm5siskpk5wp49i3gfdx5xfyaqsmqs7-nix-search-2ec97c08f0e397906e6ba6878729cab988bf2c17.drv' failed with exit code 1;
       last 10 log lines:
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > Running phase: buildPhase
       > Building subPackage ./cmd/internal/commoncmd
       > Building subPackage ./cmd/nix-dump-index
       > Building subPackage ./cmd/nix-search
       > # libdb.so/nix-search/search/searchers/blugesearcher
       > search/searchers/blugesearcher/indexer.go:108:14: undefined: unix.Renameat2
       > search/searchers/blugesearcher/indexer.go:108:77: undefined: unix.RENAME_EXCHANGE
       For full logs, run 'nix log /nix/store/0gm5siskpk5wp49i3gfdx5xfyaqsmqs7-nix-search-2ec97c08f0e397906e6ba6878729cab988bf2c17.drv'.
       ```

I was able to fix the build issue on macOS (#6), but now getting a runtime error. 
Could be my configuration: 
```
./result/bin/nix-search --index
2024-07-10T00:46:33.055-0400 [INFO]  first run or outdated index detected, will index packages
2024-07-10T00:46:33.055-0400 [INFO]  indexing packages
2024-07-10T00:46:51.077-0400 [ERROR] error: err="failed to get package index: failed to parse packages dump: EOF"
```

d443b85 needs #6 to compile without the `undefined` errors. With the #6 commit cherry picked on top, nix-search outputs the following (for invocation `nix run . -- -v apackage`
```
2024-07-11T03:39:43.630-0400 [INFO]  first run or outdated index detected, will index packages
2024-07-11T03:39:43.630-0400 [INFO]  indexing packages
2024-07-11T03:39:43.630-0400 [DEBUG] search.IndexPackages: indexing packages: nixpkgs=<nixpkgs> parallelism=10
2024-07-11T03:39:43.630-0400 [DEBUG] search.IndexPackages: worker: indexing: attrs=""
2024-07-11T03:39:43.630-0400 [DEBUG] search.IndexPackages: queued job: attrs=[]
2024-07-11T03:39:46.550-0400 [DEBUG] search.IndexPackages: done indexing packages
2024-07-11T03:39:46.550-0400 [ERROR] error: err="failed to get package index: failed to parse packages dump: EOF"
```

--- a/flake.nix
+++ b/flake.nix
@@ -24,7 +24,7 @@
 				version = self.rev or "unknown";
 				src = self;
 
-				vendorHash = "sha256-h4NVsdJksMkVai1fPc6nV9SouD3DK4UMYl70G9JJw4g=";
+				vendorHash = "sha256-gdqTTc1YsO3feN+OBeBh6inrHfZvp/dio/TUC/Aaol0=";
 
 				meta = with pkgs.lib; {
 					description = "A better and channel-compatible `nix search` for NixOS using only stable Nix tools.";

--- a/go.mod
+++ b/go.mod
@@ -8,7 +8,7 @@ require (
 	github.com/hashicorp/go-hclog v1.4.0
 	github.com/pkg/errors v0.9.1
 	github.com/urfave/cli/v3 v3.0.0-alpha2
-	golang.org/x/sys v0.8.0
+	golang.org/x/sys v0.22.1-0.20240709145142-4cafe5c66f05
 	golang.org/x/term v0.8.0
 	modernc.org/sqlite v1.8.8
 )

--- a/search/searchers/blugesearcher/indexer.go
+++ b/search/searchers/blugesearcher/indexer.go
@@ -8,7 +8,6 @@ import (
 
 	"github.com/blugelabs/bluge"
 	"github.com/hashicorp/go-hclog"
-	"golang.org/x/sys/unix"
 	"libdb.so/nix-search/search"
 )
 
@@ -105,7 +104,7 @@ func swapDir(basedir, oldname, newname string) error {
 		return err
 	}
 
-	return unix.Renameat2(int(dir.Fd()), oldname, int(dir.Fd()), newname, unix.RENAME_EXCHANGE)
+	return Rename(dir, oldname, newname)
 }
 
 func cleanOldIndexFolders(indexPath string) error {

--- a/search/searchers/blugesearcher/rename_darwin.go
+++ b/search/searchers/blugesearcher/rename_darwin.go
@@ -1,0 +1,14 @@
+//go:build darwin
+// +build darwin
+
+package blugesearcher
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+func Rename(dir *os.File, oldname, newname string) error {
+	return unix.RenameatxNp(int(dir.Fd()), oldname, int(dir.Fd()), newname, unix.RENAME_SWAP)
+}

--- a/search/searchers/blugesearcher/rename_linux.go
+++ b/search/searchers/blugesearcher/rename_linux.go
@@ -1,0 +1,14 @@
+//go:build !darwin
+// +build !darwin
+
+package blugesearcher
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+func Rename(dir *os.File, oldname, newname string) error {
+    return unix.Renameat2(int(dir.Fd()), oldname, int(dir.Fd()), newname, unix.RENAME_EXCHANGE)
+}

